Classes/Package.php

```diff
@@ -21,23 +23,52 @@ public function boot(Bootstrap $bootstrap)
         $dispatcher = $bootstrap->getSignalSlotDispatcher();
 
         $dispatcher->connect(Node::class, 'nodePropertyChanged', function (NodeInterface $node, $propertyName, $_, $newValue) use ($bootstrap) {
-            if ($propertyName !== 'identifier' || empty($newValue) || !$node->getNodeType()->isOfType('Neos.Form.Builder:IdentifierMixin')) {
+            if ($propertyName !== 'identifier' || empty($newValue) || !$node->getNodeType()->isOfType(self::NODE_TYPE_IDENTIFIER_MIXIN)) {
                 return;
             }
 
-            /** @noinspection PhpUndefinedMethodInspection */
-            $flowQuery = (new FlowQuery([$node]))->context(['invisibleContentShown' => true, 'removedContentShown' => true, 'inaccessibleContentShown' => true]);
-            $possibleIdentifier = $initialIdentifier = $newValue;
-            $i = 1;
-            /** @noinspection PhpUndefinedMethodInspection */
-            while ($flowQuery
-                    ->closest('[instanceof Neos.Form.Builder:NodeBasedForm]')
-                    // [identifier=".."] matches the Form Element identifier, [_identiier!="..."] excludes the current node
-                    ->find(sprintf('[instanceof Neos.Form.Builder:IdentifierMixin][identifier="%s"][_identifier!="%s"]', $possibleIdentifier, $node->getIdentifier()))
-                    ->count() > 0) {
-                $possibleIdentifier = $initialIdentifier . '-' . $i++;
+            $this->setUniqueFormElementIdentifier($node, $newValue);
+        });
+
+        $dispatcher->connect(Node::class, 'nodeAdded', function (NodeInterface $node) use ($bootstrap) {
```

Comment:
`$bootstrap` is not used by the closure:
```suggestion
        $dispatcher->connect(Node::class, 'nodeAdded', function (NodeInterface $node) {
```